src/Orchard.Web/Modules/Orchard.DynamicForms/ValidationRules/UrlAddress.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Orchard.DynamicForms.Helpers;
+using Orchard.DynamicForms.Services;
+using Orchard.DynamicForms.Services.Models;
+using Orchard.Localization;
+
+namespace Orchard.DynamicForms.ValidationRules
+{
+    public class UrlAddress : ValidationRule
+    {
+        public UrlAddress() {
+            RegexOptions = RegexOptions.Singleline | RegexOptions.IgnoreCase;
+            Pattern = @"^http(s)?://([\w-]+.)+[\w-]+(/[\w- ./?%&=])?$";
+        }
+
+        public string Pattern { get; set; }
+        public RegexOptions RegexOptions { get; set; }
+        
+        public override void Validate(ValidateInputContext context)
+        {
+            if (!Regex.IsMatch(context.AttemptedValue, Pattern, RegexOptions))
+            {
+                var message = GetValidationMessage(context);
+                context.ModelState.AddModelError(context.FieldName, message.Text);
+            }
+        }
+
+        public override void RegisterClientAttributes(RegisterClientValidationAttributesContext context)
+        {
+            context.ClientAttributes["data-val-regex"] = GetValidationMessage(context).Text;
+            context.ClientAttributes["data-val-regex-pattern"] = Pattern;
+        }
+
+        private LocalizedString GetValidationMessage(ValidationContext context)
+        {
+            return T(Tokenize(ErrorMessage.WithDefault(String.Format("{0} is not a valid url.", context.FieldName)), context));
```

Comment:
Yep I pulled it from the email field class.  Most of them are that way.

On Thursday, October 8, 2015, Sébastien Ros <notifications@github.com>
wrote:

> In
> src/Orchard.Web/Modules/Orchard.DynamicForms/ValidationRules/UrlAddress.cs
> <https://github.com/OrchardCMS/Orchard/pull/5884#discussion_r41557522>:
>
> > +            if (!Regex.IsMatch(context.AttemptedValue, Pattern, RegexOptions))
> > +            {
> > +                var message = GetValidationMessage(context);
> > +                context.ModelState.AddModelError(context.FieldName, message.Text);
> > +            }
> > +        }
> > +
> > +        public override void RegisterClientAttributes(RegisterClientValidationAttributesContext context)
> > +        {
> > +            context.ClientAttributes["data-val-regex"] = GetValidationMessage(context).Text;
> > +            context.ClientAttributes["data-val-regex-pattern"] = Pattern;
> > +        }
> > +
> > +        private LocalizedString GetValidationMessage(ValidationContext context)
> > +        {
> > +            return T(Tokenize(ErrorMessage.WithDefault(String.Format("{0} is not a valid url.", context.FieldName)), context));
>
> This is wrong. If you got the code from another existing class then this
> is also wrong and needs to be fixed. You can't translate a dynamic string.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OrchardCMS/Orchard/pull/5884/files#r41557522>.
>